src/Orleans/Core/ClusterClient.cs

```diff
@@ -0,0 +1,324 @@
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentException(nameof(configuration));
+            }
+
+            this.Configuration = configuration;
+            this.runtimeClient = new OutsideRuntimeClient(configuration);
+        }
+
+        /// <inheritdoc />
+        public bool IsInitialized => this.state == LifecycleState.Started;
+
+        /// <inheritdoc />
+        public IGrainFactory GrainFactory => this.InternalGrainFactory;
+
+        /// <inheritdoc />
+        internal IInternalGrainFactory InternalGrainFactory
+        {
+            get
+            {
+                this.ThrowIfDisposedOrNotInitialized();
+                return this.runtimeClient.InternalGrainFactory;
+            }
+        }
+
+        /// <inheritdoc />
+        public Logger Logger
+        {
+            get
+            {
+                this.ThrowIfDisposedOrNotInitialized();
+                return this.runtimeClient.AppLogger;
+            }
+        }
+
+        /// <inheritdoc />
+        public IServiceProvider ServiceProvider => this.runtimeClient.ServiceProvider;
+
+        /// <inheritdoc />
+        public TimeSpan ResponseTimeout
+        {
+            get { return this.runtimeClient.GetResponseTimeout(); }
+
+            set { this.runtimeClient.SetResponseTimeout(value); }
+        }
+
+        /// <inheritdoc />
+        public Action<InvokeMethodRequest, IGrain> ClientInvokeCallback
+        {
+            get { return this.runtimeClient.ClientInvokeCallback; }
+            set { this.runtimeClient.ClientInvokeCallback = value; }
+        }
+
+        /// <inheritdoc />
+        public ClientConfiguration Configuration { get; }
+
+        /// <inheritdoc />
+        IStreamProviderRuntime IInternalClusterClient.StreamProviderRuntime => this.runtimeClient.CurrentStreamProviderRuntime;
+
+        /// <inheritdoc />
+        public IEnumerable<IStreamProvider> GetStreamProviders()
+        {
+            this.ThrowIfDisposedOrNotInitialized();
+            return this.runtimeClient.CurrentStreamProviderManager.GetStreamProviders();
+        }
+
+        /// <inheritdoc />
+        public IStreamProvider GetStreamProvider(string name)
+        {
+            this.ThrowIfDisposedOrNotInitialized();
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            return this.runtimeClient.CurrentStreamProviderManager.GetProvider(name) as IStreamProvider;
+        }
+
+        /// <inheritdoc />
+        public event ConnectionToClusterLostHandler ClusterConnectionLost
+        {
+            add
+            {
+                this.ThrowIfDisposed();
+                this.runtimeClient.ClusterConnectionLost += value;
+            }
+
+            remove
+            {
+                this.ThrowIfDisposed();
+                this.runtimeClient.ClusterConnectionLost -= value;
```

Comment:
I'm not sure what the norm is, but as of the latest commit, this event is no longer public (I kept it on `IInternalClusterClient`). I've removed the throw.